tests/unit/s2n_client_finished_test.c

```diff
@@ -61,12 +61,15 @@ int main(int argc, char **argv)
         client_conn->actual_protocol_version = S2N_TLS12;
 
         /* Mutate valid verify_data */
-        POSIX_GUARD(s2n_prf_client_finished(server_conn));
+        POSIX_GUARD_RESULT(s2n_prf_client_finished(server_conn));
         server_conn->handshake.client_finished[0]++;
 
         EXPECT_SUCCESS(s2n_client_finished_send(client_conn));
         EXPECT_SUCCESS(s2n_stuffer_copy(&client_conn->handshake.io, &server_conn->handshake.io,
                 s2n_stuffer_data_available(&client_conn->handshake.io)));
+        while(s2n_conn_get_current_message_type(server_conn) != CLIENT_FINISHED) {
+            server_conn->handshake.message_number++;
+        }
```

Comment:
How does this work? Wouldn't conn->handshake.handshake_type not index to a valid array that includes CLIENT_FINISHED?